src/qt/optionsmodel.cpp

```diff
@@ -85,6 +85,21 @@ void OptionsModel::Init(bool resetSettings)
     // If SoftSetArg() or SoftSetBoolArg() return false we were overridden
     // by command-line and show this in the UI.
 
+
+    // Migration of default values
+    // Check if the QSettings container was already loaded with this client version
+    static const char strSettingsVersionKey[] = "nSettingsVersion";
+    if (!settings.contains(strSettingsVersionKey) || settings.value(strSettingsVersionKey) < CLIENT_VERSION)
+    {
+        // -dbcache was bumped from 100 to 300 in 0.13
+        // see https://github.com/bitcoin/bitcoin/pull/8273
+        // force people to upgrade to the new value if they are using 100MB
+        if (settings.contains("nDatabaseCache") && settings.value("nDatabaseCache") == 100)
```

Comment:
Would it be possible to treat a value of 0 (on disk) as "default", and
whenever a value equal to the current software default is to be written,
write 0 instead?

That would simplify future changes...